### PR TITLE
Ensure Ancestors on Correct Device During Sampling

### PR DESCRIPTION
**Pull request by lapp0**
Fixes https://github.com/outlines-dev/outlines/issues/649

### CODE REVIEW

**Comment by rlouf** [comment]
You would need to do this for every sampling method

**Comment by lapp0** [comment]
Good catch. Beam and greedy needs changes as well. 

Smoke tested, can confirm for all 3 samplers the device is set correctly:

```
>>> next_token_ids.device, ancestors.device, weights.device
(device(type='cuda', index=0), device(type='cuda', index=0), device(type='cuda', index=0))
```

File: outlines/samplers.py [hunk 1]
```diff
@@ -66,7 +66,9 @@ def __call__(
         logprobs = torch.nn.functional.log_softmax(next_token_logits, dim=-1)
         next_token_ids = torch.argmax(logprobs, dim=-1, keepdim=True)
 
-        ancestors = torch.arange(next_token_logits.shape[0])
+        ancestors = torch.arange(
+            next_token_logits.shape[0], device=next_token_logits.device
```
Comment (rlouf) [inline outlines/samplers.py:70]: nit: let's be consistent across samplers and use `next_token_logits` for each sampler :)

**Comment by rlouf** [comment]
Thanks for the fix!

**Pull request merged**